mesheryctl/internal/cli/root/config/config.go

```diff
@@ -322,6 +347,71 @@ func AddTokenToConfig(token Token, configPath string) error {
 	return nil
 }
 
+// DeleteTokenFromConfig deletes a token passed to it to mesheryctl config file
+func DeleteTokenFromConfig(tokenName string, configPath string) error {
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
```

Comment:
Can we write a generic function GetConfig(configPath string), which returns MesheryCtlConfig?
I see this code is common to all sub commands.

Reply:
Initially I was doing almost something like that. But navendu asked me to avoid extra functions. So I removed those functions.

Reply:
@navendu-pottekkat, The code from line:352-365 has repeated 4 times in this file. It would be good to convert it into a function as per [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
But it would require some rework to AddTokenToConfig , DeleteTokenFromConfig and SetTokenToConfig functions. i.e. to pass MesheryCtlConfig directly instead of utils.DefaultConfigPath

Let me know what you think.

Reply:
Yes, ideally we need to call this only on a top level command. I am looking into how we can prevent reading in the config file for each of the subcommands. Instead, we read the config once and then can just reference the central struct in which it will be stored. I am currently working on this refactor and it would be great if we could all come back to this holistically. I recently merged a PR that centralises writes to config file for contexts, a similar implementation could be added here as well.

We can make the changes here as @hexxdump and @Revolyssup has suggested in a separate PR as it would complicate things otherwise. If the current changes are good functionally, let's merge this.

Reply:
@navendu-pottekkat  sounds about right

Reply:
@navendu-pottekkat  Even after future refactoring, functionally things would be same, so I will get going with tests for token command.

Reply:
Yes, this should be good for now. 
I also thought about reading the file once and caching the data for subsequent use, however was concerned about the set operations. We need to revalidate the cache in those scenarios.

Reply:
yes, will you be adding tests in this PR?

Reply:
@navendu-pottekkat  I already created another branch for tests, thinking of keeping the PR's for tests separate.But I will proceed in whatever way you recommend.

Reply:
Okay, sounds good. Let's then get this merged.